jax/_src/path.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import pathlib
+
+logger = logging.getLogger(__name__)
+
+try:
+  import etils.epath as epath
+except:
+  epath = None
+
+# If etils.epath (aka etils[epath] to pip) is present, we prefer it because it
+# can read and write to, e.g., GCS buckets. Otherwise we use the builtin
+# pathlib and can only read/write to the local filesystem.
+if epath:
+  logger.debug("etils.epath found. Using etils.epath for file I/O.")
+  Path = epath.Path
+else:
+  logger.debug("etils.epath was not found. Using pathlib for file I/O.")
+  Path = pathlib.Path
```

Comment:
If you want to use other functions from pathlib or epath, you won't be able to do that right?

Reply:
Indeed. But do you?

(This appears to be all we actually use, and if we need more, we can add more later.)

Reply:
Fair enough. But maybe its as simple as returning the module itself?

```
if epath:
  pathlib = epath
```

Then we can just do: `from jax._src.path import pathlib` and everything works as expected.

LGTM anyways.. up to you!

Reply:
I'd rather keep the API minimal. But... it doesn't matter much. This is an internal API. If we need to change it later that's easy to do!